company/goals/guidelines.md

```diff
@@ -49,6 +44,27 @@ A goal's format is:
    - Write your goals so they communicate effectively to an audience who understands that goals aren't promises. Don't write goals so cautiously that they are hard to understand. (For example, "Improve code intelligence support for Java" is a good goal, especially with well defined outcomes. But "Investigate feasibility of improvements to Java code intelligence" is probably bad.)
 1. If a goal's success criteria can be reduced to a single quantitative metric, that's great. But don't try to force it. It's OK to rely on human judgment to judge the success of a goal.
 
+## Goal formats
+
+A goal's general format is:
+
+> **Goal title** \
+> Description of the goal and how to evaluate whether we met the goal (with a link to an automatically updated metrics dashboard, if appropriate).
+
+### Engineering team goals 
+
+Each engineering team should have its own goals page with explicit goals. Because engineering team goals are larger than a single person's goal, we use the following format: 
+
+> **Goal title:** _The title for a vision that is achievable within a 6-12 month time frame._ 
+> 
+> **Problem:** _A prose description of the problem that the goal is meant to address. The "why" of the goal._  
+>
+> **Milestones:** _Checkpoints along a linear prerequisite path that would demonstrate incremental progress towards this goal. Milestones are not specific single tasks and likely encompass 1-2 months of progress. It may make sense to list milestone-specific outcomes beneath each milestone._ 
+>
+> **Outcomes:** _The quantitative and qualitative metrics that demonstrate the goal has successfully solved the problem. Outcomes are not "tasks completed;" they are signals that the problem itself is solved._
```

Comment:
I have a sense that outcomes should be tied to milestones. Otherwise, I worry there is duplication in the best case and a disconnect in the worst cast, between the actual milestones and the outcomes we are trying to achieve.

Reply:
I agree. I actually prefer a case where they are more tightly coupled – I'll update the description/format so milestones have outcomes after I play with it a bit on my teams' pages to make sure the template formatting looks good.

Reply:
I've inlined outcomes into milestones in this template now. I'm not sure if "multiple outcomes" is better than just having a longer single outcome (example – "code insights drives 5 new sales AND 3 existing customers pay more to use code insights" vs listing those things as separate outcomes under a milestone) so perhaps we want to change that allowance. 

I also wonder if it might now make sense to give a format template and now move descriptions out of the sections or if it's still clear enough (visually) as to what a goals page should look like as is; I have no preference and see advantages to both. 

For an example of this, you can see [the PR for the code insights team page](https://github.com/sourcegraph/about/pull/2140/files#diff-851a820fbd3b799a3881f56e2de520c2e7b0e508387185d0312230e32dd315fbR28) that I've changed to now inline outcomes beneath milestones.

Reply:
I like the multiple outcomes per milestone. I think the current doc is clear and we don't need a separate template doc. Updates to code insights goals/milestones look great!